feat: add mobile version

Yinkusadel opened the pull request:
### This RR add the mobile design of TO-DO-APP, enhancing user interface and experience.

**CHANGES MADE**

-  Enhanced the ToDo list layout for better readability
- Designed visually appealing task cards for improved task representation
- Utilized Tailwind CSS classes to improve the visual presentation of tasks.

**Here is the link to the projects  [requirement page](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/02-js-todo-app.md#2-mobile-design)**

| Current Design                          | Frontend Mentor Design                               |
| --------------------------------------- | ---------------------------------------------------- |
| ![image](https://github.com/Yinkusadel/js-to-do-app/assets/105943931/61f2e42b-5219-4b41-bd65-bbbae367cc5c) | ![image](https://github.com/Yinkusadel/js-to-do-app/assets/105943931/18de29f3-e49e-4972-b8d0-959e8791f4f2) |

Yinkusadel committed:
feat: add mobile version

Yinkusadel committed:
Merge branch 'master' into feature-add-mobile-version

teckcipher commented on index.html:
```diff
@@ -6,8 +6,170 @@
-  <body class="m-0 flex min-h-screen min-w-80 place-items-center">
-    <div class="mx-auto my-0 max-w-7xl p-8 text-center" id="app"></div>
+  <body class="bg-gray-300">
+    <header class="bg-[url('/mobile-blue-bg.png')] bg-cover bg-no-repeat">
```
Let's move this to tailwind config.

teckcipher commented on index.html:
```diff
@@ -6,8 +6,170 @@
-    <div class="mx-auto my-0 max-w-7xl p-8 text-center" id="app"></div>
+  <body class="bg-gray-300">
+    <header class="bg-[url('/mobile-blue-bg.png')] bg-cover bg-no-repeat">
+      <div class="mb-0 ml-auto mr-auto mt-0 w-[30rem] pb-12 pl-6 pr-6 pt-12">
```
Can we use `w-96` instead?

teckcipher replied:
Let's use shorthand please. `mx`, `my`

teckcipher replied:
`pt-12`

teckcipher commented on index.html:
```diff
@@ -6,8 +6,170 @@
+        </section>
+
+        <form action="#">
+          <div class="font-custom flex items-center justify-between rounded bg-gray-100">
```
Can we use a more descriptive name other than `custom`.

Yinkusadel committed:
feat: update checkbox and font style

Yinkusadel committed:
refactor: checkboxex and color style

moshoodabdullahi commented on index.html:
```diff
@@ -3,11 +3,206 @@
+          </button>
+        </section>
+
+        <form action="#">
```
@Yinkusadel Nice work But I don't seem to understand the work of this (#) right here

Yinkusadel replied:
> @Yinkusadel Nice work But I don't seem to understand the work of this (#) right here

the action attribute  is used to specify the URL to which the form data should be submitted so i left the # there since i am not using any endpoint yet

moshoodabdullahi reviewed (changes requested):
Please check this out

Yinkusadel committed:
refactor: use more of shorthand property

Yinkusadel committed:
refactor: check mark

Yinkusadel committed:
refactor: update readme and make use of semantics

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Josefin+Sans:ital,wght@1,300&family=League+Spartan:wght@700&display=swap"
```
Do we need the other font used here ? `League+Spartan`?

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
Let's customise this logo.

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
-    <div class="mx-auto my-0 max-w-7xl p-8 text-center" id="app"></div>
+  <body class="bg-blue-100">
+    <header class="bg-headerimage bg-cover bg-no-repeat">
+      <div class="mx-auto my-0 w-96 px-6 py-12">
```
Can we confirm that the `my-0` is not redundant? I disabled it on the Dev console and it didn't make any different.

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+    <header class="bg-headerimage bg-cover bg-no-repeat">
+      <div class="mx-auto my-0 w-96 px-6 py-12">
+        <section class="mb-8 flex items-center justify-between">
+          <h1 class="text-2xl font-semibold tracking-[7px] text-blue-0">TODO</h1>
```
For the tracking, can we use a more responsive unit? See https://tailwindcss.com/docs/letter-spacing for recommended.

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+      <div class="mx-auto my-0 w-96 px-6 py-12">
+        <section class="mb-8 flex items-center justify-between">
+          <h1 class="text-2xl font-semibold tracking-[7px] text-blue-0">TODO</h1>
+          <button type="button" title="theme" class="cursor-pointer bg-transparent">
+            <img src="./src/images/icon-moon.svg" alt="Theme icon" />
+          </button>
```
Can we make this a checkbox instead? Such that checked and unchecked with be either of the moon or the sun icon. cc @moshoodabdullahi

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+          </button>
+        </section>
+
+        <form action="#">
```
Since there is no `action` value (for now), we can safely remove this `action` attribute.

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+
+        <form action="#">
+          <div class="flex items-center justify-between rounded bg-blue-0 font-josefin">
+            <input
+              type="checkbox"
+              name="input-name"
+              title="input"
+              class="peer relative m-0 ml-4 h-5 w-6 cursor-pointer appearance-none rounded-full border border-blue-200 from-blue-700 to-blue-800 text-blue-0 before:absolute before:left-1 checked:border-none checked:bg-gradient-to-br checked:bg-cover checked:bg-no-repeat checked:before:content-['\2713']"
+            />
+            <input
+              title="task-to-do"
+              type="text"
+              class="w-full rounded bg-blue-0 p-4 font-semibold text-blue-600 outline-none"
+              name="task-name"
+              placeholder="Create a new todo..."
+            />
```
Let's add an `arial-label` attribute to all the input element.

teckcipher commented on index.html:
```diff
@@ -3,11 +3,200 @@
+      </div>
+    </header>
+    <main>
+      <div class="m-0 mx-auto my-0 w-96 px-6 py-12 font-josefin">
```
Let's remove the `m-0` and all occurence of margin and padding zero

Yinkusadel committed:
refactor: html
implement theme toggle
add aria label to inputs
make input width fit with a lot of content
